### PR TITLE
CORE BRANCH - Update shaders to core

### PR DESCRIPTION
**Pull request by jherico**
Includes additional shaders in the test code and moves the shader testing to a new test project.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: b4bb9730e8b0daeab491a536e0b0408ef5d8829d available at: https://gist.github.com/e3a051869c2178bb999a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5474.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5474.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: cc2fed5166680a1e697406e47fb55e53d7d71b04 available at: https://gist.github.com/e3a051869c2178bb999a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5474.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5474.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3529700326fdf67d839a989683ef3bbeb386d0e5 available at: https://gist.github.com/e3a051869c2178bb999a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5474.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5474.dmg

File: libraries/gpu/src/gpu/DrawColoredTexture.slf [hunk 1]
```diff
@@ -15,8 +15,9 @@
 uniform sampler2D colorMap;
 uniform vec4 color;
 
-varying vec2 varTexcoord;
+in vec2 varTexcoord;
```
Comment (samcake) [inline libraries/gpu/src/gpu/DrawColoredTexture.slf]: need the _texcoord

File: libraries/gpu/src/gpu/GLBackend.cpp [hunk 2]
```diff
@@ -89,13 +93,6 @@ void GLBackend::init() {
 #endif
 
 #if defined(Q_OS_LINUX)
-        GLenum err = glewInit();
```
Comment (samcake) [inline libraries/gpu/src/gpu/GLBackend.cpp]: The glew init have to happen on linux too
So these deleted lines need to come  back :)

File: libraries/gpu/src/gpu/GLBackend.cpp [hunk 3]
```diff
@@ -510,6 +506,7 @@ void Batch::_glLineWidth(GLfloat width) {
     DO_IT_NOW(_glLineWidth, 1);
 }
 void GLBackend::do_glLineWidth(Batch& batch, uint32 paramOffset) {
-    glLineWidth(batch._params[paramOffset]._float);
+    // FIXME CORE
+    //glLineWidth(batch._params[paramOffset]._float);
```
Comment (samcake) [inline libraries/gpu/src/gpu/GLBackend.cpp]: this is ok, it s supported in core

File: libraries/gpu/src/gpu/GLBackendInput.cpp [hunk 4]
```diff
@@ -57,25 +57,11 @@ void GLBackend::do_setInputBuffer(Batch& batch, uint32 paramOffset) {
...
-static const int NUM_CLASSIC_ATTRIBS = Stream::TANGENT;
-static const GLenum attributeSlotToClassicAttribName[NUM_CLASSIC_ATTRIBS] = {
-    GL_VERTEX_ARRAY,
-    GL_NORMAL_ARRAY,
-    GL_COLOR_ARRAY,
-    GL_TEXTURE_COORD_ARRAY
-};
-#endif
-#endif
+#define SUPPORT_VAO
```
Comment (samcake) [inline libraries/gpu/src/gpu/GLBackendInput.cpp]: I d say leave the GLBackendINput.cpp as is, that s what i m looking at and there was slight changes

File: libraries/render-utils/src/SkyFromAtmosphere.slf [hunk 5]
```diff
@@ -51,7 +51,8 @@ uniform vec3 v3LightPos;
 uniform float g;
 uniform float g2;
 
-varying vec3 position;
+in vec3 position;
```
Comment (samcake) [inline libraries/render-utils/src/SkyFromAtmosphere.slf]: need varying prefix

File: libraries/render-utils/src/SkyFromAtmosphere.slv [hunk 6]
```diff
@@ -33,32 +33,33 @@
...
+uniform float fScale;            // 1 / (fOuterRadius - fInnerRadius)
+uniform float fScaleDepth;        // The scale depth (i.e. the altitude at which the atmosphere's average density is found)
+uniform float fScaleOverScaleDepth;    // fScale / fScaleDepth
 
 
 const int nSamples = 2;
 const float fSamples = 2.0;
 
-varying vec3 position;
+out vec3 position;
```
Comment (samcake) [inline libraries/render-utils/src/SkyFromAtmosphere.slv]: need varying prefix

File: libraries/render-utils/src/SkyFromSpace.slf [hunk 7]
```diff
@@ -53,63 +54,65 @@ uniform float g2;
 const int nSamples = 2;
 const float fSamples = 2.0;
 
-varying vec3 position;
+in vec3 position;
```
Comment (samcake) [inline libraries/render-utils/src/SkyFromSpace.slf]: need varying prefix

File: libraries/render-utils/src/SkyFromSpace.slv [hunk 8]
```diff
@@ -33,16 +33,17 @@
...
 
+<@include gpu/Inputs.slh@>
 <@include gpu/Transform.slh@>
 <$declareStandardTransform()$>
 
-uniform float fOuterRadius;		// The outer (atmosphere) radius
+uniform float fOuterRadius;        // The outer (atmosphere) radius
 
-varying vec3 position;
+out vec3 position;
```
Comment (samcake) [inline libraries/render-utils/src/SkyFromSpace.slv]: need varying prefix

File: tests/render-utils/src/main.cpp [hunk 9]
```diff
@@ -38,58 +38,9 @@
 
 #include <PathUtils.h>
 
-
 #include "gpu/Batch.h"
 #include "gpu/Context.h"
 
-#include "../model/Skybox_vert.h"
```
Comment (samcake) [inline tests/render-utils/src/main.cpp]: Did you meant to remove this code ?

File: tests/render-utils/src/main.cpp [hunk 10]
```diff
@@ -252,46 +202,6 @@ void QTestWindow::draw() {
     glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
     glViewport(0, 0, _size.width() * devicePixelRatio(), _size.height() * devicePixelRatio());
 
-    static std::once_flag once;
```
Comment (samcake) [inline tests/render-utils/src/main.cpp]: Did you mean to remove this code ?

Comment (samcake) [inline tests/render-utils/src/main.cpp]: Ah ok yes i see you created a new test

File: tests/render-utils/src/main.cpp (hunk 9 above)
Comment (samcake) [inline tests/render-utils/src/main.cpp]: Ok you created a new test

**Comment by samcake** [comment]
I d say there are 3 blocker for merge:
- GLew on linux has been removed
- Leave GLBackendInput.cpp intact because i m dealing with that 
- THe varying prefix is sometime "_" sometime "var", as we discussed yesterday we just need to pick one
 
Cheers

File: libraries/gpu/src/gpu/GLBackend.cpp (hunk 2 above)
Comment (jherico) [inline libraries/gpu/src/gpu/GLBackend.cpp]: Look at the #if above.  it covers linux and windows.

File: libraries/gpu/src/gpu/GLBackend.cpp (hunk 3 above)
Comment (jherico) [inline libraries/gpu/src/gpu/GLBackend.cpp]: It's supported in core only if the width is <= 1.0.  Otherwise it generates an error.

**Comment by jherico** [comment]
we agreed we can agree on a naming convention and fix the shaders in a later card.  Right now I just want things to render.

**Comment by hifi-gustavo** [comment]
Build console output for commit: a7dda976945d495b870501903c790f66734ab9fe available at: https://gist.github.com/e3a051869c2178bb999a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5474.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5474.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d3a9286cc26e0cd548427ba803e90d3e195d2ed8 available at: https://gist.github.com/e3a051869c2178bb999a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5474.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5474.dmg

**Comment by samcake** [comment]
You need to have the status not failing so i can merge

**Comment by hifi-gustavo** [comment]
Build console output for commit: b84eef93c99e328002415067aecd0f956a1f47e6 available at: https://gist.github.com/e3a051869c2178bb999a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5474.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5474.dmg

**Pull request merged**